flowmachine/flowmachine/features/subscriber/iterative_median_filter.py

```diff
@@ -0,0 +1,92 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from typing import List
+
+from flowmachine.core import Query
+
+
+class IterativeMedianFilter(Query):
+    """
+    Applies an iterative median filter
+
+    Parameters
+    ----------
+    query_to_filter : Query
+        Query to apply iterated median filter to.
+    filter_window_size : int
+        Size of filter window - must be odd, positive and more than 1.
+    column_to_filter : str, default 'value'
+        Column to apply the filter to.
+    partition_column : str, default 'subscriber'
```

Comment:
As far as I can see there is no test which calculates results and _includes_ the `partition_column` and `order_column`. Would be good to add one, not just to test the functionality but also as documentation on how to use it / how the result differs.